src/main/java/com/example/sortener/config/SecurityConfig.java

```diff
@@ -0,0 +1,54 @@
+package com.example.sortener.config;
+
+import com.example.sortener.security.AuthentificationProvider;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.crypto.factory.PasswordEncoderFactories;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+
+@Configuration
+public class SecurityConfig {
+
+    public SecurityConfig(AuthentificationProvider authProvider) {
+        this.authProvider = authProvider;
+    }
+
+    @Bean
+    public PasswordEncoder passwordEncoder() {
+        return PasswordEncoderFactories.createDelegatingPasswordEncoder();
+    }
+
+    private final AuthentificationProvider authProvider;
+
+    @Bean
+    public AuthenticationManager authManager(HttpSecurity http) throws Exception {
+        var authenticationManagerBuilder =
+                http.getSharedObject(AuthenticationManagerBuilder.class);
+        authenticationManagerBuilder.authenticationProvider(authProvider);
+        return authenticationManagerBuilder.build();
+    }
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        http.csrf()
+                .disable()
+                .httpBasic()
+                .disable()
+                .authorizeRequests()
+                .antMatchers("/help", "/account")
```

Comment:
```suggestion
                .antMatchers("/help", "/account").permitAll()
```
you have other matchers with permitAll() on 1 line.This one not, so fix it do marii